File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if not root:
            return []

        queue = deque()
        res = 1
        if root.left: queue.append((root.left, max(root.left.val, root.val)))
        if root.right: queue.append((root.right, max(root.right.val, root.val)))


        while queue:
            n = len(queue)

            for _ in range(n):
                node, maxx = queue.popleft()
                if node:
                    if maxx <= node.val:
                        res += 1

                    if node.left: queue.append((node.left, max(maxx, node.val)))
                    if node.right: queue.append((node.right, max(maxx, node.val)))

        return res

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func rightSideView(root *TreeNode) []int {
    if root == nil {
        return []int {}
    }

    queue := make([]*TreeNode, 0)
    queue = append(queue, root)
    res := []int {}

    for len(queue) > 0 {
        n := len(queue)

        for i := 0; i < n; i++  {
            node := queue[0]
            queue = queue[1:]

            if i == n-1 {
                res = append(res, node.Val)
            }

            if node.Left != nil {
                queue = append(queue, node.Left)
            }

            if node.Right != nil { 
                queue = append(queue, node.Right) 
            }
        }
    }

    return res
}
